src/AzureAuth/PublicClientAuth.cs

```diff
@@ -37,26 +41,46 @@ public PublicClientAuth(ILogger logger, IEnv env, ITelemetryService telemetrySer
         }
 
         /// <inheritdoc/>
-        public TokenResult Token(Guid client, Guid tenant, IEnumerable<string> scopes, AuthMode[] authModes, string domain, string prompt, TimeSpan timeout)
+        public TokenResult Token(Guid client, Guid tenant, IEnumerable<string> scopes, IEnumerable<AuthMode> authModes, string domain, string prompt, TimeSpan timeout, EventData eventData)
         {
             var result = this.tokenFetcher.AccessToken(
                 this.logger,
                 client,
                 tenant,
                 scopes,
-                authModes.Combine().PreventInteractionIfNeeded(this.env),
+                authModes.Combine().PreventInteractionIfNeeded(this.env, this.logger),
                 domain,
                 PromptHint.Prefixed(prompt),
                 timeout);
 
+            // Report individual auth flow telemetry
             result.Attempts.SendTelemetry(this.telemetryService);
 
-            if (result.Success == null)
+            var totalErrorCount = result.Attempts.SelectMany(attempt => attempt.Errors).Count();
+            eventData.Add("error_count", totalErrorCount);
+            eventData.Add("authflow_count", result.Attempts.Count);
+
+            var authflow = result.Success;

```

Comment:
This is not obvious how we set. May be add a comment why `result.Success` should be set if we had a successful authflow outcome?

Reply:
Can you elaborate on what you mean here, @shalinikhare27? I think the documentation for `TokenResult` should indicate when/why this field is set on success.

Reply:
Hmm, so though docs on `MsalWrapper.Success` describe how this is the first successful auth flow result or null. It is intentional that at this level we don't know how it's set specifically, just that we either have a success or we do not.

This is a LINQ computation, so we don't want to call it multiple times. Yes, it could be cached - but that brings with it mutability concerns and is more complicated that I think is worth implementing now.